ui/src/app/views/home-page/component.ts

```diff
@@ -50,9 +51,9 @@ export class HomePageComponent implements OnInit {
   ) {}
   ngOnInit(): void {
     this.workspacesLoading = true;
-    this.workspacesService
-        .getWorkspaces()
-        .retry(2)
+
+    retryApi(this.workspacesService
```

Comment:
I think the pattern of wrapping everything in `retryApi` is good/fine. If we could inject it into our services automatically, that'd be convenient, but this is easy & I like that it's explicit.